server/src/main/java/org/elasticsearch/cluster/metadata/DataStream.java

```diff
@@ -101,15 +101,15 @@ public DataStream removeBackingIndex(Index index) {
     }
 
     /**
-     * Generates the name of the index that conforms to the naming convention for backing indices
+     * Generates the name of the index that conforms to the default naming convention for backing indices
      * on data streams given the specified data stream name and generation.
      *
      * @param dataStreamName name of the data stream
      * @param generation generation of the data stream
      * @return backing index name
      */
-    public static String getBackingIndexName(String dataStreamName, long generation) {
-        return String.format(Locale.ROOT, "%s-%06d", dataStreamName, generation);
+    public static String getDefaultBackingIndexName(String dataStreamName, long generation) {
```

Comment:
Maybe also add a unit test just for this method? This method is used in many unit / integ tests to validate the naming of backing indices and these tests wouldn't fail if an unintended was made to this method. In later stage the rest or docs test would fail.

Reply:
@martijnvg, is the unit test in https://github.com/elastic/elasticsearch/pull/57721/commits/ec48a9532c979d88b90ffcd8d9f1b50a5175e096 along the lines of what you were suggesting?

Reply:
yes